src/index.ts

```diff
@@ -20,18 +21,105 @@ async function sendMessage() {
     })
 }
 
-// gets our copilot stream instance, and holds our buttons.
+// gets our copilot stream instance.
 async function main() {
     CopilotStreamController.getInstance();
+    const modelProgressController = ModelProgressController.getInstance();
```

Comment:
feel free to abstract some of this stuff before merging

Reply:
Going to create an issue around this abstraction for later this week and open it up for someone else to work on as well. 👍